test/integration/protocol_integration_test.cc

```diff
@@ -188,6 +193,10 @@ name: health_check
 
 // Verifies behavior for https://github.com/envoyproxy/envoy/pull/11248
 TEST_P(ProtocolIntegrationTest, AddBodyToRequestAndWaitForIt) {
+  // QUICHE can't guarantee headers and FIN to be delivered together, so
+  // headers-only request can't be detected at L7 filters.
+  EXCLUDE_DOWNSTREAM_HTTP3;
```

Comment:
Fine to leave this disabled for now but I think we can adapt the test filter to change a request with no body (headers, fin) to do (headers, fin, body).  I've adapted a few filters in my test PR so might as well wait for that to land.